src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5697,9 +5690,25 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
             ImmutableArray<TypeSymbol> typeArguments,
             DiagnosticBag diagnostics)
         {
+            bool nodeHasOmittedTypeArgs = false;
+
+            if (node.Kind() == SyntaxKind.SimpleMemberAccessExpression)
+            {
+                var memberAccessSyntax = (MemberAccessExpressionSyntax)node;
+                if (memberAccessSyntax.Name.Kind() == SyntaxKind.GenericName)
+                {
+                    var genericNameSyntax = (GenericNameSyntax)memberAccessSyntax.Name;
+                    if (genericNameSyntax.TypeArgumentList.Arguments.Any(SyntaxKind.OmittedTypeArgument))
+                    {
+                        nodeHasOmittedTypeArgs = true;
+                    }
+                }
+            }
+
             int arity;
             LookupOptions options;
-            if (typeArguments.IsDefault)
+
+            if (typeArguments.IsDefault || nodeHasOmittedTypeArgs)
```

Comment:
> || nodeHasOmittedTypeArgs [](start = 39, length = 26)

This feels suspicious. I think lookup should still respect requested arity even when there are omitted arguments.